File: app/src/main/java/com/example/geosolve/dataModel/Node.java
package com.example.geosolve.dataModel;

public class Node {
    private float x, y;
    private boolean move;

    public float getX() {
        return x;
    }

    public float getY() {
        return y;
    }

    public Node(float x, float y) {
        this.x = x;
        this.y = y;
    }

    public void moveNode(float x, float y) {
        this.x = x;
        this.y = y;
    }

    public boolean inRadius(float x, float y) {
        boolean xBool = this.x - 25 < x && x < this.x + 25;
        boolean yBool = this.y - 25 < y && y < this.y + 25;
        move = xBool && yBool;
        return move;
    }

    public boolean isMove() {
        return move;
    }

    public boolean stopMove() {
        if (move) {
            move = false;
            return true;
        }else
            return false;
    }
}

File: app/src/main/java/com/example/geosolve/model/figure/Node.kt
package com.example.geosolve.model.figure

import android.graphics.Region

class Node(var x: Float, var y: Float) {
    var isMove = false
    val neighborLines: MutableList<Line> = ArrayList()
    val neighborAngles: MutableList<Angle> = ArrayList()

    fun moveNode(x: Float, y: Float) {
        this.x = x
        this.y = y
    }

    fun inRadius(x: Float, y: Float): Boolean {
        val xBool = this.x - 25 < x && x < this.x + 25

        val yBool = this.y - 25 < y && y < this.y + 25

        return xBool && yBool
    }

    fun stopMove(): Boolean {
        return if (isMove) {
            isMove = false
            true
        } else false
    }

}